expr/putter.go

```diff
@@ -82,31 +76,31 @@ func (p *Proc) eval(ectx expr.Context, this *zed.Value) []zed.Value {
 	return vals
 }
 
-// A step is a recursive data structure encoding a series of steps to be
+// A putStep is a recursive data structure encoding a series of steps to be
 // carried out to construct an output record from an input record and
 // a slice of evaluated clauses.
-type step struct {
-	op        op
+type putStep struct {
+	op        putOp
 	index     int
 	container bool
-	record    []step // for op == record
+	record    []putStep // for op == record
 }
 
-func (s *step) append(step step) {
+func (s *putStep) append(step putStep) {
 	s.record = append(s.record, step)
 }
 
-type op int
+type putOp int
 
 const (
-	fromInput  op = iota // copy field from input record
-	fromClause           // copy field from put assignment
-	record               // recurse into record below us
+	putFromInput  putOp = iota // copy field from input record
+	putFromClause              // copy field from put assignment
+	putRecord                  // recurse into record below us
 )
 
-func (s step) build(in zcode.Bytes, b *zcode.Builder, vals []zed.Value) zcode.Bytes {
+func (s putStep) build(in zcode.Bytes, b *zcode.Builder, vals []zed.Value) zcode.Bytes {
```

Comment:
These things occupy six words, so use a pointer receiver.
```suggestion
func (p *putStep) build(in zcode.Bytes, b *zcode.Builder, vals []zed.Value) zcode.Bytes {
```